azerothcore/azerothcore-wotlk
# feat(Core/Spell): AssertSpellInfo

(cherry picked from commit https://github.com/TrinityCore/TrinityCore/commit/e52878b6b5600c2ed6fe8a60fbc27d220f807a2e)
(cherry picked from commit https://github.com/TrinityCore/TrinityCore/commit/8c44259fae2980598980a7935a3f3941130a5a10)
Co-Authored-By: Shauren <shauren.trinity@gmail.com>
Co-Authored-By: Giacomo Pozzoni <giacomopoz@gmail.com>

needed for https://github.com/azerothcore/azerothcore-wotlk/pull/6109



## Changes Proposed:
-  Use SpellMgr::AssertSpellInfo() if the spell id is valid and always supposed to return a valid not-NULL SpellInfo* 

## Tests Performed:

- none 


## How to Test the Changes:


Test that spells work as they should, you can check Blood Queen encounter that she use SPELL_PACT_OF_THE_DARKFALLEN_DAMAGE


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
